### PR TITLE
feat: adds editor functionality to CodeDisplay

### PR DESCRIPTION
**Pull request by noahamac**
Replaces `AceEditor` with `@looker/code-editor` for all code display within RunIt and ApiExplorer.

Search pattern marking:
image

Editor:
image

Response body:
image

Code tab:
image

Type declaration:
image

### CODE REVIEW

File: packages/run-it/src/components/ShowResponse/responseUtils.tsx [hunk 1]
```diff
@@ -113,7 +116,7 @@ const ShowImage = (response: IRawResponse) => {
 
 /** A handler for HTTP type responses */
 const ShowHTML = (response: IRawResponse) => (
-  <CodeDisplay language={'html'} code={response.body.toString()} />
+  <CodeDisplay language={'html'} code={response.body.toString()} transparent />
```
Comment (josephaxisa) [inline packages/run-it/src/components/ShowResponse/responseUtils.tsx:119]:
```suggestion
  <CodeDisplay language="html" code={response.body.toString()} transparent />
```

File: yarn.lock [hunk 2]
```diff
@@ -2193,6 +2193,15 @@
     debug "^2.2.0"
     es6-promise "^4.2.8"
 
+"@looker/components-providers@^0.9.29":
```
Comment (bryans99) [inline yarn.lock:2196]: Hm, how'd this get pulled in?

File: yarn.lock [hunk 3]
```diff
@@ -2205,11 +2214,38 @@
...
+  resolved "https://registry.yarnpkg.com/@looker/components-test-utils/-/components-test-utils-0.9.29.tgz#5e61437806c9a26c2d4a4759eb2ca406eee8e73d"
+  integrity sha512-m+JJ9rmvveWJRGF3vZRQo8cI35dTac1gW5pkEbs67pa5uIasyfTfI/kDWDiW7NbBClWX9aVlnpl3elsGnbt2Lw==
+
 "@looker/components-test-utils@^1.1.3":
   version "1.1.3"
   resolved "https://registry.yarnpkg.com/@looker/components-test-utils/-/components-test-utils-1.1.3.tgz#53ac11522cc7d485178c1a1d89f0751d8af88475"
   integrity sha512-z6fqkdTWw5YLcVd1FOSovl4y7W/Jsjc6GzA8sdeIMn9ghDMeH9GyGvoFIGV2tEQ/b6x1gZ+zDnKBSe8vUo4Fjg==
 
+"@looker/components@^0.9.30":
+  version "0.9.30"
```
Comment (bryans99) [inline yarn.lock:2228]: Ditto

File: packages/code-display/src/CodeDisplay.tsx [hunk 4]
```diff
@@ -112,4 +123,22 @@ export const CodeDisplay: FC<PrismEditorProps> = ({
       )}
     </Highlight>
   )
+  if (!onChange) {
+    return highlight(code)
```
Comment (josephaxisa) [inline packages/code-display/src/CodeDisplay.tsx:127]: is this for when `CodeDisplay` is used as an editor? If so I think a comment would be handy

File: packages/code-display/src/CodeDisplay.tsx [hunk 5]
```diff
@@ -36,20 +37,28 @@ import { getPrismLanguage, getOverridenTheme } from './utils'
...
   language?: string
   /** Code blob to be highlighted */
   code: string
   /** Pattern to be search (if applicable) */
   pattern?: string
+  /** onChange event handler, for Editor functionality */
+  onChange?: (text: string) => void
+  /** whether or not to render transparent background */
+  transparent?: boolean
+  height?: string
```
Comment (josephaxisa) [inline packages/code-display/src/CodeDisplay.tsx:51]: Add a comment perhaps describing what happens if no height is specified? I don't see this prop being used anywhere

File: packages/code-display/src/CodeDisplay.tsx (hunk 4 above)
Comment (josephaxisa) [inline packages/code-display/src/CodeDisplay.tsx:127]: Also L117 can be modified to use the Span from @looker/components

File: packages/code-display/src/CodeDisplay.tsx [hunk 6]
```diff
@@ -112,4 +123,22 @@ export const CodeDisplay: FC<PrismEditorProps> = ({
       )}
     </Highlight>
   )
+  if (!onChange) {
+    return highlight(code)
+  }
+  return (
+    <Editor
```
Comment (jkaster) [inline packages/code-display/src/CodeDisplay.tsx:130]: I think we should have separate components for Display and Editing. I'm presuming `Editor` is heavier than `Highlight`

**Review by jkaster — CHANGES_REQUESTED** [review]
Make CodeDisplay and CodeEditor distinct components

File: packages/code-display/src/CodeDisplay.tsx (hunk 4 above)
Comment (noahamac) [reply packages/code-display/src/CodeDisplay.tsx:127]: We can't use Span as it has default styles which conflict with Prism's styles applied onto a span

**Comment by bryans99** [comment]
I removed a reference to an old looker components-test-utils from yarn.lock

**Comment by jkaster** [comment]
Noah is working on completely removing references to Ace for this PR, so we'll hold off on approval for now

**Comment by joeldodge79** [comment]
I wonder if we should just submit a separate PR to unbreak `main` first?

**Comment by jkaster** [comment]
Is main still broken?

File: packages/code-editor/src/CodeDisplay.tsx [hunk 7]
```diff
@@ -74,22 +79,21 @@ const LineContent = styled(Span)`
...
 }) => {
-  const highlighterLang = getPrismLanguage(language)
   return (
     <Highlight
       {...defaultProps}
       code={code}
-      language={highlighterLang}
-      theme={getOverridenTheme()}
+      language={getPrismLanguage(language)}
+      theme={getOverridenTheme(transparent)}
```
Comment (jkaster) [inline packages/code-editor/src/CodeDisplay.tsx:96]: I don't know what the thinking is on calling util functions like this in the markdown. Probably fine. I typically establish the values as consts (or use state if they can change from user interaction) but defer to @bryans99 or @josephaxisa on this topic

File: packages/code-editor/src/utils/utils.ts [hunk 8]
```diff
@@ -57,7 +57,13 @@ export const getPrismLanguage = (language: string): Language => {
  * applies package overrides to the default vsCode prism theme
  * @returns modified prism theme object
  */
-export function getOverridenTheme() {
-  prismTheme.plain.backgroundColor = theme.colors.text
+export function getOverridenTheme(transparent: boolean) {
+  if (!transparent) {
```
Comment (jkaster) [inline packages/code-editor/src/utils/utils.ts:61]: hugely minor point: typically you put the positive test first, then the negative goes in the else. Some linters will recommend that.

File: yarn.lock [hunk 9]
```diff
@@ -3470,35 +3470,6 @@ accepts@~1.3.4, accepts@~1.3.5, accepts@~1.3.7:
     mime-types "~2.1.24"
     negotiator "0.6.2"
 
-ace-builds@^1.4.11, ace-builds@^1.4.6:
-  version "1.4.12"
-  resolved "https://registry.yarnpkg.com/ace-builds/-/ace-builds-1.4.12.tgz#888efa386e36f4345f40b5233fcc4fe4c588fae7"
-  integrity sha512-G+chJctFPiiLGvs3+/Mly3apXTcfgE45dT5yp12BcWZ1kUs+gm0qd3/fv4gsz6fVag4mM0moHVpjHDIgph6Psg==
```
Comment (jkaster) [inline yarn.lock:3476]: Look at those glorious deletions!

**Review by jkaster — DISMISSED** [review]
LGTM. Nice to bid Ace adieu!

File: packages/code-editor/src/CodeDisplay.tsx (hunk 7 above)
Comment (noahamac) [reply packages/code-editor/src/CodeDisplay.tsx:96]: Yeah, curious to know the standard on this.

**Review by jkaster — APPROVED** [review]

**Comment by noahamac** [comment]
What a ride. Thanks for the help, everyone :smile_cat:

**Pull request merged**